packages/compass-generative-ai/src/atlas-ai-service.ts

```diff
@@ -192,17 +190,85 @@ export function validateAIAggregationResponse(
   }
 }
 
+const aiURLConfig = {
+  // There are two different sets of endpoints we use for our requests.
+  // Down the line we'd like to only use the admin api, however,
+  // we cannot currently call that from the Atlas UI. Pending CLOUDP-251201
+  'admin-api': {
+    'user-access': (userId: string) => `unauth/ai/api/v1/hello/${userId}`,
+    aggregation: 'ai/api/v1/mql-aggregation',
+    query: 'ai/api/v1/mql-query',
+  },
+  cloud: {
+    'user-access': (userId: string) => `ai/v1/hello/${userId}`,
+    aggregation: (groupId: string) => `ai/v1/groups/${groupId}/mql-aggregation`,
+    query: (groupId: string) => `ai/v1/groups/${groupId}/mql-query`,
+  },
+} as const;
+type AIEndpoint = 'user-access' | 'query' | 'aggregation';
+
 export class AtlasAiService {
   private initPromise: Promise<void> | null = null;
 
-  constructor(
-    private atlasService: AtlasService,
-    private preferences: PreferencesAccess,
-    private logger: Logger
-  ) {
+  private apiURLPreset: 'admin-api' | 'cloud';
+  private atlasService: AtlasService;
+  private preferences: PreferencesAccess;
+  private logger: Logger;
+
+  constructor({
+    apiURLPreset,
+    atlasService,
+    preferences,
+    logger,
+  }: {
+    apiURLPreset: 'admin-api' | 'cloud';
+    atlasService: AtlasService;
+    preferences: PreferencesAccess;
+    logger: Logger;
+  }) {
+    this.apiURLPreset = apiURLPreset;
+    this.atlasService = atlasService;
+    this.preferences = preferences;
+    this.logger = logger;
+
     this.initPromise = this.setupAIAccess();
   }
 
+  private getUrlForEndpoint(
+    urlId: AIEndpoint,
+    connectionInfo?: ConnectionInfo
+  ) {
+    if (this.apiURLPreset === 'cloud') {
+      if (urlId === 'user-access') {
+        return this.atlasService.cloudEndpoint(
+          aiURLConfig[this.apiURLPreset][urlId](
+            this.preferences.getPreferencesUser().id
+          )
+        );
+      }
+
+      const atlasMetadata = connectionInfo?.atlasMetadata;
+      if (!atlasMetadata) {
+        throw new Error(
+          "Can't perform generative ai request: atlasMetadata is not available"
+        );
+      }
```

Comment:
Small nit: I guess we never expect the requests with the cloud preset to be done when metadata is missing, so maybe this check be moved to the top of the if to make this clearer